### PR TITLE
doc/orchestrator: adapt to new syntax, add more examples

### PR DESCRIPTION
**Pull request by jschmid1**
Signed-off-by: Joshua Schmid <jschmid@suse.de>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: doc/mgr/orchestrator.rst [hunk 1]
```diff
@@ -88,58 +88,134 @@ filtered to a particular host:
...
+or::
+
+    ceph orch apply osd --use-all-devices [--preview]
+
+
+For a more in-depth guide to DriveGroups please refer to :ref TODO: MAKE REF
+
+Example::
+
+    # master:~ # ceph orch daemon add osd node1:/dev/vdd
```
Comment (jschmid1) [inline doc/mgr/orchestrator.rst:128]: streamline cli syntax - omit `master:#` and use `# ceph ...`

File: doc/mgr/orchestrator.rst [hunk 2]
```diff
@@ -88,58 +88,134 @@ filtered to a particular host:
...
 
-Where ``drive.group.json`` is a JSON file containing the fields defined in
-:class:`ceph.deployment_utils.drive_group.DriveGroupSpec`
+
+or::
+
+    ceph orch apply osd --use-all-devices [--preview]
+
+
+For a more in-depth guide to DriveGroups please refer to :ref TODO: MAKE REF
```
Comment (jschmid1) [inline doc/mgr/orchestrator.rst:124]: TODO

**Comment by jschmid1** [comment]
jenkins render docs

**Comment by jschmid1** [comment]
Ready for review.
I'll open the discussion around DriveGroup vs OSDSpec again and would argue that we can't just s/DriveGroup/OSDSpec.

My current feeling is that DriveGroups are subspec (as implemented) of an OSDSpec and should be treated as such. If we talk about the part that is being passed to cephadm, it's the OSDSpec. When talking about device filters, limiters etc, we should still refer to DriveGroups. wdyt?

File: doc/mgr/orchestrator.rst [hunk 3]
```diff
@@ -70,8 +70,8 @@ List hosts associated with the cluster::
 
 Add and remove hosts::
 
-    ceph orch host add <host>
-    ceph orch host rm <host>
+    orch host add <hostname> [<addr>] [<labels>...]
```
Comment (sebastian-philipp) [inline doc/mgr/orchestrator.rst:73]:
```suggestion
    ceph orch host add <hostname> [<addr>] [<labels>...]
```

File: doc/mgr/orchestrator.rst [hunk 4]
```diff
@@ -88,58 +88,134 @@ filtered to a particular host:
...
+    node2   /dev/vdc  hdd   8192M  228094  False  LVM detected, Insufficient space (<5GB) on vgs, locked
 
-    Host 192.168.121.181:
-    Device Path           Type       Size    Rotates  Available Model
-    /dev/sdb               hdd      50.0G       True       True ATA/QEMU HARDDISK
-    /dev/sda               hdd      50.0G       True      False ATA/QEMU HARDDISK
 
 .. note::
-    Output form Ansible orchestrator
+    Output form Cephadm orchestrator
```
Comment (sebastian-philipp) [inline doc/mgr/orchestrator.rst:105]: as all `device ls` come from ceph-volume nowadays, I'd remove this note.

File: doc/mgr/orchestrator.rst [hunk 5]
```diff
@@ -88,58 +88,134 @@ filtered to a particular host:
...
 
-Decommission an OSD
+Remove an OSD
 -------------------
 ::
 
-    ceph orch osd rm <osd-id> [osd-id...]
+    orch osd rm <svc_id>... [--replace] [--force]
```
Comment (sebastian-philipp) [inline doc/mgr/orchestrator.rst:150]:
```suggestion
    ceph orch osd rm <svc_id>... [--replace] [--force]
```

File: doc/mgr/orchestrator.rst [hunk 6]
```diff
@@ -88,58 +88,134 @@ filtered to a particular host:
...
 
-Removes one or more OSDs from the cluster and the host, if the OSDs are marked as
-``destroyed``.
+Removes one or more OSDs from the cluster.
 
 Example::
 
     # ceph orch osd rm 4
```
Comment (sebastian-philipp) [inline doc/mgr/orchestrator.rst:156]: for Pacific, we have to correct the order to `ceph orch rm osd 4`

File: doc/mgr/orchestrator.rst [hunk 7]
```diff
@@ -88,58 +88,134 @@ filtered to a particular host:
...
+    osd.5 node3 3 2020-04-22 19:28:34.201685
+    osd.3 node2 0 2020-04-22 19:28:34.201695
+
+
+When no PGs are left on the osd, it will be decommissioned and removed from the cluster.
+
 
 .. note::
-    Output form Ansible orchestrator
+    Output form Cephadm orchestrator
```
Comment (sebastian-philipp) [inline doc/mgr/orchestrator.rst:175]: I'd remove this. Rook will use the same code.

**Comment by sebastian-philipp** [comment]
> My current feeling is that DriveGroups are subspec (as implemented) of an OSDSpec and should be treated as such. If we talk about the part that is being passed to cephadm, it's the OSDSpec. When talking about device filters, limiters etc, we should still refer to DriveGroups. wdyt?

Where does a **user** need to know the difference between an OSDSpec and a Drive Group? by specifying 

```yaml
service_type: osd
```
We've basically already introduced the name OSDSpec, not Drive Group. maybe we should make it even more clear by using the `spec:` prefix:


```yaml
service_type: osd
placement: {"host_pattern": "*"}
spec:
  data_devices:
    all: true
```

To make it clear, that the OSDSpec needs some additional properties, like data_devices etc?

File: doc/mgr/orchestrator.rst (hunk 6 above)
Comment (jschmid1) [reply doc/mgr/orchestrator.rst:156]: do we? why is that?

Comment (sebastian-philipp) [reply doc/mgr/orchestrator.rst:156]: either we agree on 

```
ceph orch <verb> <objec> params
```
or on

```
ceph orch <objec>  <verb> params
```
Right now we have an evil mix of both.

**Pull request merged**